### ISSUE TITLE
Add Lua 5.2 support

### ISSUE THREAD

**Issue by BradleyChatha** [BradleyChatha/lumars#7]
I am currently also over in my own fork over to Lua 5.2 due to needing proper support for lua_yield and lua_resume. Which 5.1 doesn't have, while also needing goto which 5.2 has.

_Originally posted by @LunaTheFoxgirl in https://github.com/BradleyChatha/lumars/issues/6#issuecomment-1014327570_

**Comment by LunaTheFoxgirl** [comment]
Any updates on this? Going to be using lumars in Inochi Session for handling plugin scripting, and Lua 5.2 (and luajit) support would be very much appreciated

**Comment by LunaTheFoxgirl** [comment]
I just realised that LuaJIT only supports 5.1, so I guess I'd just want the other features for injecting functions in to tables

**Comment by BradleyChatha** [comment]
Ah, if it's a priority now then I'll get onto it ASAP.
Will hopefully get something done tomorrow for you :)

**Comment by BradleyChatha** [comment]
Got that done for you, let me know if you have any other issues that you urgently need done.

Love your work with your Inochi software, so I'm happy to help where I can

**Comment by ethindp** [comment]
Would you mind adding 5.3 and 5.4 support as well? I love this library and am going to be using it in most likely a few projects -- this has to be one of the most mature Lua bindings for D I've seen.

**Comment by BradleyChatha** [comment]
I wouldn't mind at all. The only real blockers are the ever fleeting amounts of time; energy, and motivation :D

I still need to get 5.2 properly done though, I might try to make more progress this week, but no promises 👼

**Comment by ethindp** [comment]
Yeah, I get that, sometimes I struggle with motivation to work on things lol.
